ktor-server/ktor-server-core/jvm/src/io/ktor/server/engine/EmbeddedServerJvm.kt

```diff
@@ -388,3 +388,9 @@ actual constructor(
         }
     }
 }
+
+internal fun checkUrlMatches(url: URL, pattern: String): Boolean {
+    val urlPath = url.path?.replace(File.separatorChar, '/') ?: return false
+    val normalizedPattern = pattern.replace(File.separatorChar, '/')
+    return urlPath?.contains(normalizedPattern, ignoreCase = true) == true
```

Comment:
One more minor change
```suggestion
    return urlPath.contains(normalizedPattern, ignoreCase = true)
```